catalog/spatial/registry/registry-report-action-provider/src/main/java/org/codice/ddf/registry/report/action/provider/RegistryReportActionProvider.java

```diff
@@ -0,0 +1,224 @@
+        }
+
+        try {
+            String registryId =
+                    URLEncoder.encode(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                            .getValue()
+                            .toString(), (StandardCharsets.UTF_8).toString());
+            String sourceId = URLEncoder.encode(getSource(metacard),
+                    (StandardCharsets.UTF_8).toString());
+            Action action = getAction(registryId, sourceId);
+
+            if (action == null) {
+                return Collections.emptyList();
+            }
+
+            return Collections.singletonList(action);
+
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.debug("Unsupported Encoding exception", e);
+            return Collections.emptyList();
+        }
+    }
+
+    private List<Action> processSubject(Source source) {
+        String configId = source.getId();
+        if (configId != null) {
+            Configuration[] configurations;
+            try {
+
+                configurations = configurationAdmin.listConfigurations(String.format("(id=%s)",
+                        configId));
+
+                if (configurations.length > 0) {
+                    for (Configuration configuration : configurations) {
+                        if (configuration.getProperties()
+                                .get(RegistryObjectMetacardType.REGISTRY_ID) != null) {
+                            return processSubject(configurations[0]);
+                        }
+                    }
+                }
+            } catch (IOException | InvalidSyntaxException e) {
+                LOGGER.debug("Unable to access source configurations", e);
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    private List<Action> processSubject(Configuration configuration) {
+        if (configuration.getProperties()
+                .get(RegistryObjectMetacardType.REGISTRY_ID) != null) {
+            try {
+                String registryId = URLEncoder.encode(configuration.getProperties()
+                        .get(RegistryObjectMetacardType.REGISTRY_ID)
+                        .toString(), (StandardCharsets.UTF_8).toString());
+                Action action = getAction(registryId, "");
+
+                if (action == null) {
+                    return Collections.emptyList();
+                }
+
+                return Collections.singletonList(action);
+            } catch (UnsupportedEncodingException e) {
+                LOGGER.debug("Unsupported Encoding exception", e);
+                return Collections.emptyList();
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    private Action getAction(String metacardId, String sourceId) {
+
+        if (StringUtils.isNotBlank(sourceId)) {
+            sourceId = SOURCE_ID_QUERY_PARAM + sourceId;
+        }
+
+        URL url;
+        try {
+
+            URI uri = new URI(SystemBaseUrl.constructUrl(
```

Comment:
Would rather see `String.format` or a `StringBuilder` used to build up this string.